community/kernel/src/main/java/org/neo4j/kernel/configuration/ConfigValues.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.configuration;
+
+import java.util.List;
+import java.util.Map;
+
+import org.neo4j.helpers.Function;
+import org.neo4j.helpers.Pair;
+
+import static java.util.stream.Collectors.toList;
+import static org.neo4j.helpers.Pair.pair;
+
+public class ConfigValues implements Function<String, String>
```

Comment:
Again, use `java.util.function.Function` instead. In fact, apart from the `rawConfiguration` method, this entire thing can be replaced with `raw::get` in the use sites, assuming the inner "raw" HashMap is available.